builtin/logical/database/rotation.go

```diff
@@ -456,6 +458,7 @@ func (b *databaseBackend) setStaticAccount(ctx context.Context, s logical.Storag
 				Statements:   statements,
 			}
 			input.Role.StaticAccount.PrivateKey = wal.NewPrivateKey
+			usedExistingCredential = true
```

Comment:
Should we also generate a new Private Key credential if one fails? Or should the fix be focussed to password specific?

Reply:
We have only had the issue reported for password credentials. But I don't think there is any harm in adding this here.